src/main/java/org/opensearch/ad/rest/RestExecuteAnomalyDetectorAction.java

```diff
@@ -115,21 +146,36 @@ private String validateAdExecutionInput(AnomalyDetectorExecutionInput input) {
     }
 
     @Override
-    public List<Route> routes() {
-        return ImmutableList.of();
-    }
-
-    @Override
-    public List<ReplacedRoute> replacedRoutes() {
+    public List<ReplacedRouteHandler> replacedRouteHandlers() {
         return ImmutableList
             .of(
                 // get AD result, for regular run
-                new ReplacedRoute(
+                new ReplacedRouteHandler(
                     RestRequest.Method.POST,
-                    String.format(Locale.ROOT, "%s/{%s}/%s", AnomalyDetectorPlugin.AD_BASE_DETECTORS_URI, DETECTOR_ID, RUN),
+                    String.format(Locale.ROOT, "%s/{%s}/%s", AnomalyDetectorExtension.AD_BASE_DETECTORS_URI, DETECTOR_ID, RUN),
                     RestRequest.Method.POST,
-                    String.format(Locale.ROOT, "%s/{%s}/%s", AnomalyDetectorPlugin.LEGACY_OPENDISTRO_AD_BASE_URI, DETECTOR_ID, RUN)
+                    String.format(Locale.ROOT, "%s/{%s}/%s", AnomalyDetectorExtension.LEGACY_OPENDISTRO_AD_BASE_URI, DETECTOR_ID, RUN),
+                    handleRequest
                 )
             );
     }
+
+    private ExtensionRestResponse getExecuteDetectorResponse(
+        RestStatus restStatus,
+        RestRequest request,
+        AnomalyResultResponse response,
+        String content
+    ) throws IOException {
+        ExtensionRestResponse extensionRestResponse;
+        if (content.isEmpty()) {
+            extensionRestResponse = new ExtensionRestResponse(
```

Comment:
You can just return from here. We can avoid having a separate variable here and in the else too.

Reply:
Changes pushed